UnifiedApi/Client/Services/Submitter/Service.cs

```diff
@@ -305,20 +305,35 @@ public string SessionId
           catch (Exception e)
           {
             errorHandler(resultStatusData.TaskId,
+                         TaskStatus.Error,
                          e.Message + e.StackTrace);
           }
         }
 
         missing.ExceptWith(resultStatusCollection.IdsReady.Select(x => x.TaskId));
 
-        foreach (var resTuple in resultStatusCollection.IdsResultError)
+        foreach (var resultStatusData in resultStatusCollection.IdsResultError)
         {
-          var outputInfo = SessionService.GetTaskOutputInfo(resTuple.TaskId);
+          var details    = "";
+          var outputInfo = SessionService.GetTaskOutputInfo(resultStatusData.TaskId);
+          var taskStatus = SessionService.GetTaskStatus(resultStatusData.TaskId);
 
-          errorHandler(resTuple.TaskId,
-                       outputInfo.TypeCase == Output.TypeOneofCase.Error
-                         ? outputInfo.Error.Details
-                         : "Result is in status : " + resTuple.Status + ", look for task in error in logs.");
+          switch (taskStatus)
+          {
+            case TaskStatus.Canceling:
+            case TaskStatus.Canceled:
+              details = $"Task {resultStatusData.TaskId} was canceled";
+              break;
+            default:
```

Comment:
outputInfo could be retrieved only in this case. Saves one request

Reply:
good point